design/service-binding.md

```diff
@@ -115,7 +115,7 @@ spec:
     resourceRef: my-cluster
 ```
 
-Then the binding information is supplied in in a standard way, such as environment variables. Of course, in combination with developer tooling, this can remove the need for the developer to write the code to read credentials and so on. Instead, they're presented to the application as environment variables with well known names.
+Then the binding information is supplied in in a standard way, such as environment variables or secrets. For convenience, I have in general chosen to use secrets for binding information (this is controlled by the CSV annotations). Of course, in combination with developer tooling, this can remove the need for the developer to write the code to read credentials and so on. Instead, they're presented to the application as environment variables with well known names.
```

Comment:
Instead of `or secrets.` in the first sentence, do you mean `or a volume mounted secret.`? Is the binding metadata in the backing service CRD or CSV also attempting to control how each binding property is presented to the client app, i.e., as an env var or as a volume mounted file? I would have expected the binding metadata in the backing service CRD or CSV to focus exclusively on populating the binding secret with the appropriate data and then let the client app via the ServiceBinding CR decide whether to consume the secret properties as environment variables or volume mounts. Since storing secrets as environment variables is widely agreed to be less secure, service binding should support consuming secrets exclusively as volume mounts. This might best be achieved by letting the client specify in the ServiceBinding CR how binding properties should be consumed.

Reply:
Thanks for your comments. I agree that it would be better to have the `ServiceBinding` CR choose how binding properties should be consumed. The syntax being used by the Service Binding Specification is evolving in tandem with this document and I'm still ironing out inconsistencies.